### PR TITLE
feat: add short links importer from bigquery

### PR DESCRIPTION
**Pull request by tataihono**
# New Feature for Importing Short Links from BigQuery

This PR implements a new feature for importing short links data from BigQuery into our database. The implementation follows our camel case naming conventions and includes comprehensive test coverage.

## Key Components

1. **Short Links Importer**
   - Created a new module for importing short links from BigQuery tables
   - Implemented schema validation using Zod to ensure data integrity
   - Added support for environment-specific hostname configuration (prd, stg, dev)

2. **Data Processing**
   - Implemented single record import (`importOne`) and batch import (`importMany`) functions
   - Added validation for URL format and timestamp fields
   - Included error handling for invalid data formats

3. **Domain Management**
   - Added functionality to automatically create or retrieve the appropriate domain based on environment
   - Implemented domain caching to improve performance during batch imports

4. **Test Coverage**
   - Added comprehensive tests for all import scenarios
   - Included tests for error conditions and edge cases
   - Mocked external dependencies for reliable test execution

## Technical Details

- The importer connects to the BigQuery table `jfp-data-warehouse.jfp_shorturls_prod.yourls_url`
- Short links are stored with environment-specific domains (prd, stg, dev)
- The implementation follows our naming conventions, using camel case for "shortLink" throughout

## Summary by CodeRabbit

* **New Features**
  * Introduced functionality to import short links into the database.
  * Added a new command-line target to queue short link imports.
  * Added a new service to manage short link import processing.

* **Bug Fixes**
  * Enhanced error handling and validation for importing short link data.

* **Tests**
  * Added comprehensive tests for short link import operations.
  * Updated service tests to verify short link import execution.

* **Chores**
  * Improved configuration to support short link import via a new project command.

### CODE REVIEW

File: apis/api-media/src/workers/bigQuery/importers/shortLinks/shortLinks.spec.ts [hunk 1]
```diff
@@ -0,0 +1,252 @@
+import { Service as PrismaService } from '.prisma/api-media-client'
```
[the remark this answers is not in the document]
Comment (tataihono) [reply apis/api-media/src/workers/bigQuery/importers/shortLinks/shortLinks.spec.ts:1]: always ignore this.

File: apis/api-media/src/workers/bigQuery/importers/shortLinks/shortLinks.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import { Logger } from 'pino'
+import { z } from 'zod'
+
+import { Service as PrismaService } from '.prisma/api-media-client'
```
[the remark this answers is not in the document]
Comment (tataihono) [reply apis/api-media/src/workers/bigQuery/importers/shortLinks/shortLinks.ts:4]: @coderabbitai always ignore this.

**Comment by edmonday** [comment]
@tataihono had to remove on stage label off, couldn't resolve merge conflict when resetting stage

**Review by mikeallisonJS — DISMISSED** [review]
